src/output.rs

```diff
@@ -17,18 +19,23 @@ pub enum OutputType {
 
 impl OutputType {
     #[cfg(feature = "paging")]
-    pub fn from_mode(mode: PagingMode, pager: Option<&str>) -> Result<Self> {
+    pub fn from_mode(paging_mode: PagingMode, wrapping_mode: WrappingMode, pager: Option<&str>) -> Result<Self> {
         use self::PagingMode::*;
-        Ok(match mode {
-            Always => OutputType::try_pager(false, pager)?,
-            QuitIfOneScreen => OutputType::try_pager(true, pager)?,
+        use self::WrappingMode::*;
+        Ok(match paging_mode {
+            Always => OutputType::try_pager(false, wrapping_mode == Character, pager)?,
```

Comment:
Let's try to not repeat ourselves here, even if its a small piece of code: I would extract `wrapping_mode == Character` into a separate variable (also: see the proposed `enum` change below).